rinja/src/html.rs

```diff
@@ -0,0 +1,203 @@
+use std::{fmt, str};
+
+#[cfg(not(any(target_arch = "x86", target_arch = "x86_64")))]
+#[allow(unused)]
+pub(crate) fn write_escaped_str(mut fmt: impl fmt::Write, string: &str) -> fmt::Result {
+    // Even though [`jetscii`] ships a generic implementation for unsupported platforms,
+    // it is not well optimized for this case. This implementation should work well enough in
+    // the meantime, until portable SIMD gets stabilized.
+
+    // Instead of testing the platform, we could test the CPU features. But given that the needed
+    // instruction set SSE 4.2 was instroduced in 2008, that it has an 99.61 % avalibility rate
+    // in Steam's June 2024 hardware survey, and is a prerequisite to run Windows 11, I don't
+    // think we need to care.
+
+    let mut escaped_buf = ESCAPED_BUF_INIT;
+    let mut last = 0;
+
+    for (index, byte) in string.bytes().enumerate() {
+        let escaped = match byte {
```

Comment:
Using a bit test in here makes the benchmarks run slower. It could be that the benchmarks outline an unrealistic scenario, though: They consist of comparatively long strings.

Bit testing is slower than a table lookup, if the table is already loaded into the CPU cache, but loading the table takes quite some time \_once\_, too. If the strings are mostly short, and mostly contain no characters that need escaping, then a bit test solution would win.

I guess the current solution is good enough for now. Just something we need to keep in mind, that we might need to revise the benchmark text corpus. Even the "short" string is 27 characters long.